src/main/java/com/minecraftabnormals/upgrade_aquatic/core/events/EntityEvents.java

```diff
@@ -146,6 +148,17 @@ public static void onEntityUpdate(LivingUpdateEvent event) {
 				}
 			}
 		}
+		if (entity instanceof DrownedEntity) {
+			Pose pose = EntityEvents.getHorizontalMotion(((DrownedEntity) entity).getMotion()) >= 0.025F && ((DrownedEntity) entity).getEntityWorld().getFluidState(((DrownedEntity) entity).getPosition().down()).isTagged(FluidTags.WATER) ? Pose.SWIMMING : Pose.STANDING;
+			if (entity.getPose() != pose)
+				((DrownedEntity) entity).setPose(pose);
+		}
+	}
+	
+	private static float getHorizontalMotion(Vector3d motion) {
```

Comment:
A similar method to this already exists in `DrownedModelMixin`.
Also, it'd be better to use `Entity.horizontalMag()` instead because the square root value of the horizontal motion is not needed.
```Java
if (Entity.horizontalMag(entity.getMotion()) >= 0.000625F)
```